ratis-server/src/main/java/org/apache/ratis/server/impl/RaftServerImpl.java

```diff
@@ -1645,7 +1647,27 @@ leaderId, getMemberId(), currentTerm, followerCommit, inconsistencyReplyNextInde
       return reply;
     });
   }
-  private CompletableFuture<Void> appendLog(List<LogEntryProto> entries) {
+
+    /**
+     * The listener checks whether it can become a follower.
+     *
+     * @return
+     */
+    private CompletableFuture<Void> checkAndUpdateListenerState() {
```

Comment:
We should move the logic to changeToFollower(..).  Then it will work for other cases such as loading a snapshot.